scalanet/src/io/iohk/scalanet/peergroup/udp/DynamicUDPPeerGroup.scala

```diff
@@ -30,24 +29,27 @@ import scodec.{Attempt, Codec}
 import scala.util.control.NonFatal
 
 /**
-  * PeerGroup implementation on top of UDP.
+  * PeerGroup implementation on top of UDP that always opens a new channel
+  * from a random port when sending a message to a remote address.
```

Comment:
I find this doc a little bit confusing.  What it describes is the way that KNetwork used this peer group. In theory nothing keeps user from creating channl per peer (each with randomly assigned port) and then using those channels to communicate with remote peer.

I would try to be more specific and say that each call to `PeerGroup.client` generates randomly assigned port to communicate with remote peer.

Reply:
You are right, that's more precise, thanks :+1: